src/charm.py

```diff
@@ -215,13 +230,44 @@ def ready_to_start(self) -> bool:
         Returns:
             True if ZK is related and `sync` user has been added. False otherwise.
         """
+        # SSL must be enabled for Kafka and ZK or disabled for both
+        if self.tls.enabled ^ (
+            self.kafka_config.zookeeper_config.get("ssl", "disabled") == "enabled"
+        ):
+            logger.error("SSL must be enabled for Zookeeper and Kafka, or disabled for both")
+            return False
```

Comment:
Overall I really like this check. Could we `block` or `waiting` here? Not sure which is most appropriate, just something for the user to see 'You need to do something here before your brokers work.